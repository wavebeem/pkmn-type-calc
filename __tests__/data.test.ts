import {
  defensiveMatchups,
  Effectiveness,
  Type,
  GroupedMatchups
} from "../src/data";

describe(".defensiveMatchups", () => {
  describe("for normal and none", () => {
    let result: GroupedMatchups;

    beforeEach(() => {
      result = defensiveMatchups(Type.NORMAL, Type.NONE);
    });

    describe(".typesFor", () => {
      let typesFor: Type[];

      describe("when effectiveness is zero", () => {
        beforeEach(() => {
          typesFor = result.typesFor(Effectiveness.ZERO);
        });

        it("should only have 1 entry", () => {
          expect(typesFor).toHaveLength(1);
        });

        it("should contain the correct items", () => {
          expect(typesFor).toMatchSnapshot();
        });
      });

      describe("when effectiveness is quarter", () => {
        beforeEach(() => {
          typesFor = result.typesFor(Effectiveness.QUARTER);
        });

        it("should be empty", () => {
          expect(typesFor).toHaveLength(0);
        });
      });

      describe("when effectiveness is half", () => {
        beforeEach(() => {
          typesFor = result.typesFor(Effectiveness.HALF);
        });

        it("should be empty", () => {
          expect(typesFor).toHaveLength(0);
        });
      });

      describe("when effectiveness is regular", () => {
        beforeEach(() => {
          typesFor = result.typesFor(Effectiveness.REGULAR);
        });

        it("should have 16 items", () => {
          expect(typesFor).toHaveLength(16);
        });

        it("should contain the correct items", () => {
          expect(typesFor).toMatchSnapshot();
        });
      });

      describe("when effectiveness is double", () => {
        beforeEach(() => {
          typesFor = result.typesFor(Effectiveness.DOUBLE);
        });

        it("should have 1 item", () => {
          expect(typesFor).toHaveLength(1);
        });

        it("should contain the correct items", () => {
          expect(typesFor).toMatchSnapshot();
        });
      });

      describe("when effectiveness is quadruple", () => {
        beforeEach(() => {
          typesFor = result.typesFor(Effectiveness.QUADRUPLE);
        });

        it("should be empty", () => {
          expect(typesFor).toHaveLength(0);
        });
      });
    });

    it("generates a grouped matchup", () => {
      expect(result).toBeInstanceOf(GroupedMatchups);
    });
  });
});
