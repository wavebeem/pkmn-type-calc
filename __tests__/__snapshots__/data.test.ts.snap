// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.defensiveMatchups for normal and none .typesFor when effectiveness is double should contain the correct items 1`] = `
Array [
  "fighting",
]
`;

exports[`.defensiveMatchups for normal and none .typesFor when effectiveness is regular should contain the correct items 1`] = `
Array [
  "normal",
  "flying",
  "poison",
  "ground",
  "rock",
  "bug",
  "steel",
  "fire",
  "water",
  "grass",
  "electric",
  "psychic",
  "ice",
  "dragon",
  "dark",
  "fairy",
]
`;

exports[`.defensiveMatchups for normal and none .typesFor when effectiveness is zero should contain the correct items 1`] = `
Array [
  "ghost",
]
`;
